baremetal_network_provisioning/tests/unit/ml2/extensions/test_bnp_switch.py

```diff
@@ -0,0 +1,179 @@
+            mock.patch.object(discovery_driver.SNMPDiscoveryDriver,
+                              'get_ports_status',
+                              return_value=ports_list)):
+            switch = self.bnp_wsgi_controller.show(show_req, switch_id)
+            db.get_bnp_switch_port_map_by_switchid.called
+            discovery_driver.SNMPDiscoveryDriver.get_ports_status.called
+            return switch
+
+    def test_create_show_switch(self):
+        switch = self._create_switch(self.data, self.bnp_switch_dict)
+        switch = switch.pop('bnp_switch')
+        switch_id = switch['id']
+        self._show_switch(switch_id)
+
+    def test_create_list_switch(self):
+        self._create_switch(self.data, self.bnp_switch_dict)
+        self._create_switch(self.data1, self.bnp_switch_dict1)
+        list_req = self.new_list_request('bnp-switches')
+        result = self.bnp_wsgi_controller.index(list_req)
+        result = result.pop('bnp_switches')
+        self.assertEqual(2, len(result))
+
+    def test_update_with_invalid_protocol(self):
+        switch = self._create_switch(self.data, self.bnp_switch_dict)
+        switch_id = switch['bnp_switch']['id']
+        data = {"bnp_switch": {"access_protocol": "snmpv4"}}
+        self.assertRaises(webob.exc.HTTPBadRequest,
+                          self._update_switch, data, switch_id)
+
+    def test_show_invalid_switch(self):
+        self._create_switch(self.data, self.bnp_switch_dict)
+        switch_id = 'foobar'
+        show_req = self.new_show_request('bnp-switches', switch_id)
+        self.assertRaises(webob.exc.HTTPNotFound,
+                          self.bnp_wsgi_controller.show, show_req,
+                          switch_id)
+
+    def test_delete_invalid_switch(self):
+        switch_id = 'foobar'
+        self.assertRaises(webob.exc.HTTPNotFound,
+                          self._delete_switch,
+                          switch_id)
+
+    def test_create_with_invalid_vendor(self):
+        data = self.data
+        data['bnp_switch']['vendor'] = 'fake_vendor'
+        self.assertRaises(webob.exc.HTTPBadRequest,
+                          self._create_switch,
+                          self.data, self.bnp_switch_dict)
+
+    def test_create_with_same_ipaddress(self):
+        self._create_switch(self.data, self.bnp_switch_dict)
+        self.assertRaises(webob.exc.HTTPConflict,
+                          self._create_switch,
+                          self.data, self.bnp_switch_dict)
+
+    def test_update_with_invalid_credentials(self):
+        switch = self._create_switch(self.data, self.bnp_switch_dict)
+        switch_id = switch['bnp_switch']['id']
+        data = {"bnp_switch": {"access_protocol": "snmpv4",
+                               "access_parameters": {
+                                   "write_community": "public"}}}
+        self.assertRaises(webob.exc.HTTPBadRequest,
+                          self._update_switch, data, switch_id)
```

Comment:
looks good to me.
nit: docstring is missing for all methods even for the methods in source file.